llvm/include/llvm/Frontend/OpenMP/OMPIRBuilder.h

```diff
@@ -607,15 +621,17 @@ class OpenMPIRBuilder {
   /// Generator for '#omp barrier'
   ///
   /// \param Loc The location where the barrier directive was encountered.
-  /// \param DK The kind of directive that caused the barrier.
+  /// \param Kind The kind of directive that caused the barrier.
   /// \param ForceSimpleCall Flag to force a simple (=non-cancellation) barrier.
   /// \param CheckCancelFlag Flag to indicate a cancel barrier return value
   ///                        should be checked and acted upon.
+  /// \param ThreadID Optional parameter to pass in any existing ThreadID value.
   ///
   /// \returns The insertion point after the barrier.
-  InsertPointTy createBarrier(const LocationDescription &Loc, omp::Directive DK,
-                              bool ForceSimpleCall = false,
-                              bool CheckCancelFlag = true);
+  InsertPointTy createBarrier(const LocationDescription &Loc,
+                              omp::Directive Kind, bool ForceSimpleCall = false,
+                              bool CheckCancelFlag = true,
+                              Value *ThreadID = nullptr);
```

Comment:
What has this to do with reductions? Either make it a pre-commit or just don't introduce this concept of "existing ThreadID values" again. Most (>90%) of the runtime functions that take a thread ID don't use it anyway. It's leftover from the old runtime.

Reply:
Most of the clang reduction tests have it in them where they reuse the ThreadID. This change allows us to pass an existing ThreadID if available.

If you'd like I could remove this change and update the tests instead, please let me know which you prefer.

Reply:
The ThreadID doesn't seem to be passed in the original code. If there was reuse before perhaps some optimization is not happening? Probably worth looking into the cause of this instead of adding the parameter.

Reply:
This entire threadId in API stuff is only a leftover from the old (broken) runtime design.
We don't use the thread Id in barriers, and we don't need to pass it in.
If that means you need to change the tests, so be it. Do it before this commit, and feel free to modify the device RTL API to remove the argument completely.